django/template/utils.py

```diff
@@ -109,3 +110,19 @@ def get_app_template_dirs(dirname):
     ]
     # Immutable return value because it will be cached and shared by callers.
     return tuple(template_dirs)
+
+
+class AltersData:
+    def __init_subclass__(cls, **kwargs):
+        mro = inspect.getmro(cls)[1:]
```

Comment:
Just use `cls.__mro__`, `inspect.getmro` seems to be useless

https://github.com/python/cpython/blob/8af04cdef202364541540ed67e204b71e2e759d0/Lib/inspect.py#L729-L731

Also I think you really only need to loop over `cls.__bases__`? https://docs.python.org/3.10/library/stdtypes.html#class.__bases__ . Because if `getattr` finds a function on a class, it exists either in that class or in its ancestors, so that covers the ancestors in the MRO.